Wire --descheduling-interval to run descheduler on a loop

damemi opened the pull request:
Fixes https://github.com/kubernetes-sigs/descheduler/issues/220

This wires the `--descheduling-interval` flag to actually run the descheduler on a loop in the interval specified. If the value is unset (which defaults to `0`) the descheduler only runs once, maintaining consistency with the current expectation. 

One benefit to this is that it removes the need to use a CronJob for a continuously running controller.

aveshagarwal commented:
> Fixes #220
> 
> This wires the `--descheduling-interval` flag to actually run the descheduler on a loop in the interval specified. If the value is unset (which defaults to `0`) the descheduler only runs once, maintaining consistency with the current expectation.
> 
> One benefit to this is that it removes the need to use a CronJob for a continuously running controller.

CronJob are widely used so not sure it helps much. So have an option where if someone uses descheduler as cronjob, they can disable it.

damemi commented:
@aveshagarwal the way I've written this, the default value for `descheduling-interval` (0s) still runs the descheduler only once, then exits. This makes the new control-loop behavior opt-in, and shouldn't see any change for current users unless. However we could make this clearer or more difficult to "accidentally" switch on if you'd like.

I think the ability to execute multiple descheduler runs in the same pod does have some other advantages too, such as easier debugging and the ability to filter through logs of multiple runs, along with simplifying deployment.

ravisantoshgudimetla commented:
@damemi - I am fine with the changes, thanks for working on this. @aveshagarwal is right that we thought about it but didn't implement since no one was asking for it.

Please update the docs string at https://github.com/kubernetes-sigs/descheduler/blob/dd54f1a6560f2945502af3b5347beaeceb1449a2/cmd/descheduler/app/options/options.go#L51, to state our intent clearly.

damemi commented:
@ravisantoshgudimetla thanks, I updated the doc string and also added an E2E which confirms that by default, the descheduler should still only run once and exit (this took some minor refactoring of our main run function)
